pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -238,11 +234,18 @@ func TestReconcile(t *testing.T) {
 	machineHealthCheck := maotesting.NewMachineHealthCheck("machineHealthCheck")
 	nodeStartupTimeout := 15 * time.Minute
 	machineHealthCheck.Spec.NodeStartupTimeout = metav1.Duration{Duration: nodeStartupTimeout}
+	machineHealthCheck.Spec.FailedNodeStartupTimeout = metav1.Duration{Duration: time.Hour}
 
 	machineHealthCheckNegativeMaxUnhealthy := maotesting.NewMachineHealthCheck("machineHealthCheckNegativeMaxUnhealthy")
 	negativeOne := intstr.FromInt(-1)
 	machineHealthCheckNegativeMaxUnhealthy.Spec.MaxUnhealthy = &negativeOne
 
+	//Failed Machine
+	machineWithFailedStatus := maotesting.NewMachine("machineWithFailedStatus", "")
+	failedVar := machinePhaseFailed
+	machineWithFailedStatus.Status.Phase = &failedVar
+	machineWithFailedStatus.CreationTimestamp = metav1.Now()
```

Comment:
Do we have a test case that covers a failed machine with and without noderef? Would be good to make sure we have both test cases (and also make the description of this more descriptive to show exactly what this is for, i.e. Failed Machine with no Node)

Reply:
Good catch.
Actually we only have a case without the noderef.
I'll add a case with a noderef (and a providerID)